spynnaker/pyNN/models/neuron/population_vertex.py

```diff
@@ -474,35 +498,195 @@ def set_structure(self, structure: BaseStructure) -> None:
         self.__structure = structure
 
     @property
-    def combined_core_capable(self) -> bool:
+    def use_combined_core(self) -> bool:
         """
-        Whether the vertex can manage to operate on a combined
+        Whether the vertex should operate on a combined
         neuron-synapse core, or if a split synapse-core is more
         appropriate.
 
-        .. note::
-            This is currently based only on the ITCM available, not
-            on the incoming synapses, but could be combined with
-            n_synapse_cores_required to determine if, and how-many, synapse
-            cores are needed.
-
         :rtype: bool
         """
-        return self.__synapse_dynamics.is_combined_core_capable
+        # If we can't use a combined core, use a split core
+        if not self.__synapse_dynamics.is_combined_core_capable:
+            return False
+
+        # If we can't use a split core, use a combined core
+        if not self.__synapse_dynamics.is_split_core_capable:
+            return True
+
+        # If the user has chosen to have a synapse core, add one
+        if self.__n_synapse_cores is not None and self.__n_synapse_cores > 0:
+            return False
+
+        # If the time-step is less than 1, use combined core if no synapse
+        # cores are needed, otherwise use split core
+        # TODO: Look at if it is possible to include neurons in a combined
+        # core calculation and update to allow a choice of combined core if
+        # neurons and synapses fit on a single core
+        if SpynnakerDataView().get_simulation_time_step_ms() < 1.0:
+            return self.n_synapse_cores_required == 0
+
+        # If the timestep is 1 or greater, use a combined core generally
+        return True
 
     @property
     def n_synapse_cores_required(self) -> int:
         """
         The estimated number of synapse cores required, when using a
         split synapse-neuron core model.
 
-        .. note::
-            This is currently hard-coded but could be updated to work
-            this out based on the number of incoming synapses.
-
         :rtype: int
         """
-        return 1
+        self.__update_n_synapse_cores()
```

Comment:
if you are always going to call self.__update...

Why not merge the two methods and just have it return the value.

for example:
 def n_synapse_cores_required(self) -> int:
   if self.__n_synapse_cores is not None:
       return self.__n_synapse_cores
  ....
  return self.__n_synapse_cores

Reply:
Updated